### PR TITLE
Developer-targeted detailed documentation of Oxli binary formats

### PR DESCRIPTION
**Pull request by kdmurray91**
Hi all,

This PR adds some personal notes I've taken from reading the source code on the binary file formats in khmer/oxli.

I'm not sure if it will be useful to the project, but here it is in any case. I'll polish it up a bit further, and maybe add some more formats, if you think it's useful.

Cheers,
K

### CODE REVIEW

**Comment by mr-c** [comment]
@kdmurray91 I am blown away, this is really wonderful

File: doc/dev/binary-file-formats.rst [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+Number of Tables    1      10   Number of Nodegraph tables.
+================== ===== ===== ==============================================
+
+Then follows the Nodegraph's tables. For each table:
+
+================== ======= ===== ==============================================
+Field               Len     Off     Value
+================== ======= ===== ==============================================
+Table size          8         0   Length of table, **in bits**. (``uint64_t``)
+Bins                N/8+1     8   This table's bytes, length given by previous field, divided by 8, plus 1. (``uint8_t``)
```
Comment (kdmurray91) [inline doc/dev/binary-file-formats.rst]: @mr-c Is there a way of wrapping lines in RST simple tables that doesn't create a new row? Or do I have to suck it up and make a proper table?

Comment (mr-c) [inline doc/dev/binary-file-formats.rst]: I think you can line wrap if you keep the correct indentation?

On Sun, Jun 14, 2015 at 10:33 PM Kevin Murray <notifications@github.com>
wrote:

> In doc/dev/binary-file-formats.rst
> <https://github.com/dib-lab/khmer/pull/1101#discussion_r32392578>:
>
> > +Field               Len   Off     Value
> > +================== ===== ===== ==============================================
> > +Magic string        4       0   ``OXLI`` (``SAVED_SIGNATURE``)
> > +Version             1       4   ``0x04`` (``SAVED_FORMAT_VERSION``)
> > +File Type           1       5   ``0x02`` (``SAVED_HASHBITS``)
> > +K-size              4       6   k-mer length, as ``unsigned int``.
> > +Number of Tables    1      10   Number of Nodegraph tables.
> > +================== ===== ===== ==============================================
> > +
> > +Then follows the Nodegraph's tables. For each table:
> > +
> > +================== ======= ===== ==============================================
> > +Field               Len     Off     Value
> > +================== ======= ===== ==============================================
> > +Table size          8         0   Length of table, **in bits**. (``uint64_t``)
> > +Bins                N/8+1     8   This table's bytes, length given by previous field, divided by 8, plus 1. (``uint8_t``)
>
> @mr-c <https://github.com/mr-c> Is there a way of wrapping lines in RST
> simple tables that doesn't create a new row? Or do I have to suck it up and
> make a proper table?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dib-lab/khmer/pull/1101/files#r32392578>.
>
-- 
Michael R. Crusoe: Programmer & Bioinformatician crusoe@ucdavis.edu
<mcrusoe@msu.edu>
The lab for Data Intensive Biology; University of California, Davis
https://impactstory.org/MichaelRCrusoe http://twitter.com/biocrusoe

Comment (kdmurray91) [inline doc/dev/binary-file-formats.rst]: So you can. emphasis on correct :smile:

**Comment by ctb** [comment]
+1

**Comment by kdmurray91** [comment]
Don't think I'll add much to this in the current round, as that's all I have written. So I'd say it ready for review, and I'll add to it later.

- [ ] Is it mergeable?
- [ ] Did it pass the tests?
- [ ] If it introduces new functionality in scripts/ is it tested?
  Check for code coverage with `make clean diff-cover`
- [ ] Is it well formatted? Look at `make pep8`, `make diff_pylint_report`,
  `make cppcheck`, and `make doc` output. Use `make format` and manual
  fixing as needed.
- [ ] Did it change the command-line interface? Only additions are allowed
  without a major version increment. Changing file formats also requires a
  major version number increment.
- [ ] Is it documented in the ChangeLog?
  http://en.wikipedia.org/wiki/Changelog#Format
- [ ] Was a spellchecker run on the source code and documentation after
  changes were made?
- [ ] Is the Copyright year up to date?

**Comment by kdmurray91** [comment]
Oh, do `RST` files in `doc/dev/` need a copyright header? Like [`getting-started.rst`](https://raw.githubusercontent.com/dib-lab/khmer/master/doc/dev/getting-started.rst)?

**Comment by mr-c** [comment]
@kdmurray91 yes, but we haven't decided on what to write there since the move. I'm okay without it for now.

File: ChangeLog [hunk 2]
```diff
@@ -1,3 +1,9 @@
+2015-06-16  Kevin Murray  <spam@kdmurray.id.au>
+
+   * doc/dev/binary-file-formats.rst: Add documentation of Khmer's binary file
```
Comment (mr-c) [inline ChangeLog]: We always spell it `khmer` to differentiate from the people group

**Comment by mr-c** [comment]
LGTM, thank you @kdmurray91

**Pull request merged**